projecttacoma/deqm-test-kit
# Inferno Template Update

# Summary
Updates ruby version to 3.1.2, inferno-core version to 0.4.5, and updates the Inferno template according to these [outlined instructions](https://github.com/inferno-framework/inferno-template/releases/tag/v0.0.3).

## New behavior
No new behavior is directly introduced in this PR. See the [inferno-core CHANGELOG.md](https://github.com/inferno-framework/inferno-core/blob/main/CHANGELOG.md) to see new behavior introduced by the inferno-core upgrade from 0.3.5 to 0.4.5.

## Code changes
Most of the code changes reflected in this PR come from applying the Inferno template update and from running `rubocop` after updating the ruby version to 3.1.2.

Some changes to pay attention to:
* The changes in `docker-compose.yml`/`docker-compose.background.yml`
* The use of the double-splat (**) operator for specifying `measure_id_args` as an input in the tests that require a `measure_id` to be specified. Without this operator, `inferno-core` mistakes the `measure_id_args` for another input name, which must be a symbol or a string (`measure_id_args` is a hash). Using the double-splat operator allows `inferno-core` to match `measure_id_args` to the hash argument in its input-gathering function.
* In `lib/utils/data_requirements_utils.rb` - function `get_data_requirements_queries()`’s parameter `include_patient` no longer has a default value - the inclusion of the default value was causing the corresponding `rspec` tests to fail (saying that 2 args were received when only 1 was expected). Changing `include_patient: false` to `include_patient = false` fixed the issue but raised rubocop issues. I figured we can remove the default value since we only use the function in a few places within the code.

Most of the remaining code changes are a result of applying the template patch or running rubocop.

# Testing guidance
* Run `rspec` tests
* Run in Docker with `docker compose pull` followed by `docker compose up --build`
* If you are running into hanging tests in the test kit, try running `docker-compose run inferno bundle exec rake db:migrate`
* Inspect the [commit that applies the patch](https://github.com/projecttacoma/deqm-test-kit/commit/aebae58ebf23af6d7aa0715145f955fba047c568) to ensure that all the changes are accurate in regard to deqm-test-kit
